File: .skills/openclaw-skills/skills/mrblackeg/tiktok-streak-bot/scripts/config_loader.py
import json
import os
import logging

class ConfigLoader:
    def __init__(self, config_path):
        self.config_path = config_path
        self.config = {}

    def load(self):
        if os.path.exists(self.config_path):
            logging.info(f"Loading configuration from {self.config_path}...")
            with open(self.config_path, 'r', encoding='utf-8') as f:
                self.config = json.load(f)
            return self.config
        else:
            logging.error(f"Configuration file not found at {self.config_path}.")
            return None

    def get(self, key, default=None):
        return self.config.get(key, default)

File: .skills/openclaw-skills/skills/brokemac79/webchat-audio-notifications/client/config-loader.js
/**
 * WebchatNotifications Config Loader
 * Load notification settings from JSON config file
 * 
 * Usage:
 * const config = await loadNotificationConfig('./config.json');
 * const notifier = new WebchatNotifications(config.notifications);
 */

async function loadNotificationConfig(configPath = './notification-config.json') {
  try {
    const response = await fetch(configPath);
    if (!response.ok) {
      throw new Error(`Config file not found: ${configPath}`);
    }
    
    const config = await response.json();
    
    // Validate config
    if (!config.notifications) {
      throw new Error('Config must have "notifications" object');
    }
    
    // Return just the notifications config
    return config.notifications;
  } catch (error) {
    console.error('Failed to load notification config:', error);
    console.log('Using default configuration');
    
    // Return defaults
    return {
      soundPath: './sounds',
      soundName: 'level3',
      defaultVolume: 0.7,
      cooldownMs: 3000,
      enableButton: true,
      debug: false
    };
  }
}

/**
 * Initialize notifications from config file
 * One-liner setup with JSON config
 * 
 * Usage:
 * const notifier = await initNotificationsFromConfig('./config.json');
 */
async function initNotificationsFromConfig(configPath = './notification-config.json') {
  const config = await loadNotificationConfig(configPath);
  const notifier = new WebchatNotifications(config);
  await notifier.init();
  return notifier;
}

// Export for module systems
if (typeof module !== 'undefined' && module.exports) {
  module.exports = {
    loadNotificationConfig,
    initNotificationsFromConfig
  };
}
